src/components/consul/selectors/ConsulDocumentSelector.js

```diff
@@ -0,0 +1,109 @@
+import PropTypes from 'prop-types';
+import React, { useState } from 'react';
+import { TouchableOpacity } from 'react-native';
+
+import { colors, Icon, normalize, texts } from '../../../config';
+import { formatSize } from '../../../helpers';
+import { DOCUMENT_TYPE_PDF, useSelectDocument } from '../../../hooks';
+import { Button } from '../../Button';
+import { Input } from '../../form';
+import { RegularText } from '../../Text';
+import { WrapperRow } from '../../Wrapper';
+
+export const DocumentSelector = ({ control, field, item, documentsAttributes }) => {
+  const [errorMessage, setErrorMessage] = useState([]);
+  const [pdfInfoText, setPDFInfoText] = useState([]);
+
+  const { buttonTitle, infoText } = item;
+  const { name, onChange, value } = field;
+
+  const { selectDocument } = useSelectDocument();
+
+  return (
+    <>
+      <Input {...item} control={control} hidden name={name} value={JSON.stringify(value)} />
+      <RegularText smallest placeholder>
+        {infoText}
+      </RegularText>
+
+      {value
+        ? JSON.parse(value).map((item, index) => (
+            <>
+              <WrapperRow key={index} center spaceBetween>
+                <RegularText>{item.title}</RegularText>
+
+                <TouchableOpacity
+                  onPress={() => {
+                    documentsAttributes.splice(index, 1);
+                    errorMessage.splice(index, 1);
+                    pdfInfoText.splice(index, 1);
+                    onChange(JSON.stringify(documentsAttributes));
+                    setErrorMessage(errorMessage);
+                    setPDFInfoText(pdfInfoText);
```

Comment:
what is going on here? can you explain please why are you using splice? and splice does mutate the state variables itself to pass them in the setter afterwards? seems bad practice for me. state variables should always be set per setter function.

so at least

```
setErrorMessage(errorMessage.splice(index, 1));
setPDFInfoText(pdfInfoText.splice(index, 1));
```

i guess

Reply:
i think we need a helper function for clarity, that receives an array and return a new array with spliced stuff?